components/support/android/src/main/java/mozilla/appservices/support/native/RustBuffer.kt

```diff
@@ -47,13 +48,18 @@ open class RustBuffer : Structure() {
 
     fun asCodedInputStream(): CodedInputStream? {
         return this.data?.let {
-            CodedInputStream.newInstance(it.getByteBuffer(0, this.len))
+            if (this.len < Int.MIN_VALUE || this.len > Int.MAX_VALUE) {
+                throw RuntimeException("len does not fit in a int")
+            }
+            CodedInputStream.newInstance(it.getByteArray(0, this.len.toInt()))
         }
     }
 
     fun asCodedOutputStream(): CodedOutputStream? {
         return this.data?.let {
-            CodedOutputStream.newInstance(it.getByteBuffer(0, this.len))
+            val method = CodedOutputStream::class.java.getDeclaredMethod("newSafeInstance", ByteBuffer::class.java)
+            method.isAccessible = true
+            return method.invoke(null, it.getByteBuffer(0, this.len)) as CodedOutputStream
```

Comment:
We do the same here, by [forcing the usage](https://github.com/protocolbuffers/protobuf/blob/e667bf6eaaa2fb1ba2987c6538df81f88500d030/java/core/src/main/java/com/google/protobuf/CodedOutputStream.java#L130-L140) of `newSafeInstance`

Reply:
Ouch :/